packages/block-library/src/navigation/edit/inner-blocks.js

```diff
@@ -73,6 +73,11 @@ export default function NavigationInnerBlocks( {
 	const showPlaceholder =
 		! hasCustomPlaceholder && ! hasMenuItems && ! isSelected;
 
+	const defaultRenderingMode = useSelect( ( select ) => {
+		const { getBlockEditingMode } = select( blockEditorStore );
+		return getBlockEditingMode( clientId ) === 'default';
+	} );
```

Comment:
I think this should be available via block edit context. In that case, there's no need to create another store subscription.